gtk/MessageLogWindow.cc

```diff
@@ -141,13 +142,30 @@ void MessageLogWindow::Impl::scroll_to_bottom()
 
 void MessageLogWindow::Impl::level_combo_init(Gtk::ComboBox* level_combo) const
 {
+    auto const current_level = gtr_pref_int_get(TR_KEY_message_level);
+    auto const default_level = TR_LOG_INFO;
+
+    auto current_row = std::optional<int>{};
+    auto default_row = int{ 0 };
+
     auto items = std::vector<std::pair<Glib::ustring, int>>{};
     for (auto const& [level, name] : level_names_)
     {
+        if (level == current_level)
+        {
+            current_row = std::size(items);
+        }
+
+        if (level == default_level)
+        {
+            default_row = std::size(items);
+        }
+
         items.emplace_back(name, level);
     }
+
     gtr_combo_box_set_enum(*level_combo, items);
-    gtr_combo_box_set_active_enum(*level_combo, gtr_pref_int_get(TR_KEY_message_level));
+    level_combo->set_active(current_row ? *current_row : default_row);
```

Comment:
I don't think breaking away from the `gtr_combo_box_set_active_enum()` abstraction is a good idea; I'd rather change its interface to either accept an [optional] default value and handle it internally
```cpp
gtr_combo_box_set_active_enum(*level_combo, gtr_pref_int_get(TR_KEY_message_level), TR_LOG_INFO);
```
or return a `bool` to for something like
```cpp
if (!gtr_combo_box_set_active_enum(*level_combo, gtr_pref_int_get(TR_KEY_message_level)))
{
    gtr_combo_box_set_active_enum(*level_combo, TR_LOG_INFO);
}
```

If we do break away though,
```suggestion
    level_combo->set_active(current_row.value_or(default_row));
```

Reply:
Ah! I always forget about value_or

Reply:
Updated in 7dc51830a91252b0f06a80c51dd78687d58ff7d6. Keeping `gtr_combo_box_set_active_enum()` allows us to drop the size_t / int mismatch too.